packages/serverpod_serialization/test/auth_encoding_test.dart

```diff
@@ -0,0 +1,156 @@
+import 'dart:convert';
+
+import 'package:test/test.dart';
+import 'package:serverpod_serialization/src/auth_encoding.dart';
+
+void main() {
+  /// conventional serverpod auth key format user:pwd
+  var conventionalAuthKeys = [
+    'joe:pwd',
+    'joe,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}\r\n : pwd,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}\r\n',
+  ];
+
+  /// standard basic auth schemes' key formats, which is also equivalent to the wrapped key format
+  var standardBasicAuthKeys = [
+    'Basic dGVzd',
+    'Basic dGVzd02348+/',
+    'Basic dGVzd==',
+  ];
+
+  /// standard non-basic auth schemes' key formats
+  var standardNonBasicAuthKeys = [
+    'Bearer dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}',
+    'Digest dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}',
+    'HOBA dGVzd ,;.<>_-/\\|"\'`~!@#\$%^&*()+=[]{}',
+  ];
+
+  /// arbitrary auth key formats
+  var arbitraryAuthKeys = [
+    '',
+    ' ',
+    '\t',
+    '\n',
+    '\r',
+    ':',
+    ':::::::::::::::',
+    ',;.:<>_-/\\|"\'`~!@#\$%^&*()+=[]{}\r\n',
+    '%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%',
+    json.encode({'key': 'value'}),
+    json.encode({
+      'key': 'value',
+      'key2': {
+        'key3': 'value3',
+        'key4': {
+          'key5': 'value5',
+          'key6': 'value6',
+          'key7': 'value7',
+          'key8': 'value8',
+          'key9': 'value9',
+          'key10': 'value10',
+        },
+      },
+      'key10': [],
+      'key11': [1, 2],
+      'key12': null,
+    }),
+  ];
+
+  group('isValidAuthHeaderValue()', () {
+    for (var key in [
+      ...standardBasicAuthKeys,
+      ...standardNonBasicAuthKeys,
+    ]) {
```

Comment:
I also think this is more or less supported in different language stacks. The Dart test package sadly doesn't have any built-in support for parameterized tests. We could add our own helper along the way if we had a strong use case. As you mention, you can then add a human readable description for each parameter (not saying we should do it here.
```dart
group("formatDay should format dates correctly:", () {
  var inputsToExpected = {
    DateTime(2018, 11, 01): "Thu 1",
    ...
    DateTime(2018, 11, 07): "Wed 7",
    DateTime(2018, 11, 30): "Fri 30",
  };
  inputsToExpected.forEach((input, expected) {
    test("$input -> $expected", () {
      expect(formatDay(input), expected);
    });
  });
});
```

Reply:
I had 2 discussions with Isak on this :)
Yes there is a point with unrolling, but for this specific kind of unit test my view would be that it is much easier to get an overview of the test, maintain the test, and get a reasonable understanding of the completeness of the test cases to list them like this. Also, otherwise one would wonder if and if so what is different in the test code of dozens of tests.

Reply:
Looking closer at this code I agree with what @christerswahn said above. I think simple validation functions like this are one of the prime use cases for parameterized tests. In my opinion, unrolling these into 6 different test cases that does the same thing will make it harder to overview what is tested, whereas now it's super quick to see what cases are covered. As always there are some downsides but in this case I think the benefits outweigh them.